### ISSUE TITLE
weak-Orthogonal Quantization

### ISSUE THREAD

**Issue by hbellafkir** [thulab/DeepHash#35]
Hi everyone,
I have some questions concerning the triplet quantization loss in DTQ method.
 1. the implementation doesn't square the quantization loss as the paper suggests. is there a reason why? 
2. the weak orthogonal constraint is also not used in the quantization loss as the paper suggests. is there a reason why ?

in model/dtq.py: 
  ```
def quantization_loss(self, z, h): 
        with tf.name_scope('quantization_loss'):
            q_loss = tf.reduce_mean(tf.reduce_sum(z - tf.matmul(h, self.C), -1))
        return q_loss
 ```

**Comment by bl0** [comment]
Thanks for your issues. 

1. I found it's a bug which I made accidentally when I'm clean the code to release. I will fix this soon.
2. The current released version of DTQ indeed does not use weak orthogonal constraint. I will add this as soon as possible.

**Comment by hbellafkir** [comment]
Hey thanks for the reply.
Suppose that the loss is correct implemented, which cq_lambda should I use to reproduce the results in the paper ? Because currently the value is set to 0 (quantization loss is not considered at all)

**Comment by bl0** [comment]
Hi, I have fixed the bug. Thanks again for your issues.

cq_lambda should be small enough such as 1e-4 in order to avoid affecting model training.

**Comment by hbellafkir** [comment]
ok thanks, I will try that and give feedback.

**Comment by hbellafkir** [comment]
hey @bl0,
I trained with your suggested cq_lambda value (1e-4) and with the other default parameters for 40 epochs on cifar10. the following observations can be made:

   1. No convergence regarding the quantization loss can be observed:
![Screenshot from 2019-06-23 10-42-58](https://user-images.githubusercontent.com/49009668/59973729-c216e900-95a3-11e9-921d-783de9295817.png).

2. The result of 0.792 MAP@54000 is not reached as the paper claims:
![Screenshot from 2019-06-23 10-49-46](https://user-images.githubusercontent.com/49009668/59973792-ab24c680-95a4-11e9-9e91-62648861454f.png)

**Comment by bl0** [comment]
Thanks for your feedback.
